paddle/fluid/operators/math/softmax_impl.h

```diff
@@ -38,16 +40,37 @@ template <typename DeviceContext, typename T, bool is_test, typename Enable>
 void SoftmaxFunctor<DeviceContext, T, is_test, Enable>::operator()(
     const DeviceContext& context, const int axis_dim,
     const framework::Tensor* X, framework::Tensor* Y) {
-  auto logits = EigenMatrix<T>::From(*X);
-  auto softmax = EigenMatrix<T>::From(*Y);
-
+  auto in_dims = X->dims();
   const int kBatchDim = 0;
   const int kClassDim = 1;
 
-  const int batch_size = logits.dimension(kBatchDim);
-  const int num_classes = logits.dimension(kClassDim);
+  const int num_classes = in_dims[kClassDim];
+  const int batch_size = in_dims[kBatchDim];
   const int num_remain = num_classes / axis_dim;
 
+  if (num_remain == 1 && platform::MayIUse(platform::avx)) {
+    const T* in_data = X->data<T>();
+    T* out_data = Y->data<T>();
+    for (int bs = 0; bs < batch_size; ++bs) {
+      T max_val = *std::max_element(in_data, in_data + num_classes);
+      max_val *= -1;
+      vec_add_bias<T, platform::avx>(num_classes, max_val, in_data, out_data);
+      vec_exp<T>(num_classes, out_data, out_data);
+
+      T sum = 0;
+      vec_sum<T, platform::avx>(num_classes, out_data, &sum);
+      sum = static_cast<T>(1) / sum;
+      vec_scal<T, platform::avx>(num_classes, sum, out_data, out_data);
+
+      in_data += num_classes;
+      out_data += num_classes;
+    }
+    return;
```

Comment:
What will happen when we have situation that we exectue softmax normalization after first axe or no AVX support. It seems there will no be any computation? Is this correct

Reply:
This only improve on avx platform and only axis=1, if not, use original implementations.

Reply:
ok